File: 프로그래머스/1/138477. 명예의 전당 （1）/명예의 전당 （1）.java
import java.util.*;
class Solution {
    public List<Integer> solution(int k, int[] score) {
        List<Integer> answer = new ArrayList<>();
        List<Integer> list = new ArrayList<>();
        
        for(int i=0;i<score.length;i++)
        {
            list.add(score[i]);
            list.sort(Comparator.reverseOrder());
            if(i<k)
            {
                answer.add(list.get(list.size()-1));
            }
            else{
            
                answer.add(list.get(k-1));
            }
            
        }
        
        return answer;
    }
}

File: 프로그래머스/1/138477. 명예의 전당 （1）/명예의 전당 （1）.js
function solution(k, score) {
    var answer = [];
    var list = [];
    for(let num of score){
        list.push(num);
        list.sort((a,b)=>b-a);
        if(list.length <k){
            answer.push(list[list.length-1]);
        }else{
            answer.push(list[k-1]);
        }
        
    }
    
    return answer;
}